### ISSUE TITLE
How to use Tensorboard with tf-agents?

### ISSUE THREAD

**Issue by ayansengupta17** [tensorflow/agents#423]
Let us take the example of SAC minitaur tutorial. Now, I would like to see the loss of actor and critic on tensorboard. What is easiest way of doing it?
Thanks in advance

**Comment by wookayin** [comment]
You can find a full example here: https://github.com/tensorflow/agents/blob/master/tf_agents/agents/sac/examples/v2/train_eval.py

Tensorboard event files will be written into `train_dir` (or `root_dir`), so you can launch tensorboard on the log directory.

**Comment by kbanoop** [comment]
Thanks @wookayin for the answer. Closing the issue, please feel free to reopen if you have more questions.

**Comment by kaiyun717** [comment]
Is there a way to view the model graph in TensorBoard using TF-agents? 
I'd like to see the entire network.

**Comment by luzgui** [comment]
Hello, Im having the same issue. I wonder if @kaiyun717 managed to solve.
I can log different metrics and different runs in tensorboard but I am not sure how to write the Q-network to the logdir

**Comment by kaiyun717** [comment]
@luzgui Hello! I checked my old code and found the way I did it. Please note, however, that this was back in January 2021, and I have not used TF-Agents since. So the code and comments I made back then may not work, but it'll give you a good idea of how to start!

(Below is my documentation back in Jan, 2021)

## Graph in Tensorboard
In Tensoboard eager mode, you need to use the `@tf.function` decorator in order to graph your model.

Wrap around a custom function that has your agent step in your environment once with `@tf.function`. 
```python
@tf.function
def agent_trace(environment, agent):
    time_step = environment.current_time_step()       
    action_step = agent.policy.action(time_step)        

graphDir = "your/graph/directory" 
writer = tf.summary.create_file_writer(graphDir)

# Commands to trace your `tf.function`'s
tf.summary.trace_on(graph=True, profiler=True)

agent_trace(environment, agent)

with writer.as_default():
    tf.summary.trace_export(name="model_trace", step=0, profiler_outdir=graphDir)
```

---
Here, you are having the agent step in the environment and tracing its "decision" process, aka the policy (neural network mostly), by wrapping a custom function with the `@tf.function` decorator. 
I believe this should do the trick. Hope it works!
